File: typescript/src/messages/verdict_request.ts
import { JsonProperty, JsonObject } from "typescript-json-serializer";
import { Kind, Message } from "./message";

@JsonObject()
export class VerdictRequest extends Message {
  public constructor(sha256: string, guid: string, session_id: string, use_cache?: boolean, use_hash_lookup?: boolean) {
    super(Kind.VerdictRequest);
    this.sha256 = sha256;
    this.session_id = session_id;
    this.guid = guid;
    this.use_cache = use_cache;
    this.use_hash_lookup = use_hash_lookup;    
  }

  @JsonProperty() public sha256: string;
  @JsonProperty() public guid: string;
  @JsonProperty() public session_id: string;
  @JsonProperty() public use_hash_lookup?: boolean;
  @JsonProperty() public use_cache?: boolean;
}

File: golang/vaas/v2/pkg/messages/verdict_request.go
// Package messages provides structures for handling communication messages between the client and the VaaS server.
package messages

import (
	"github.com/GDATASoftwareAG/vaas/golang/vaas/v2/pkg/options"
	"github.com/google/uuid"
)

// verdictRequest represents a generic verdict request.
type verdictRequest struct {
	Kind                     Kind                     `json:"kind" default:"VerdictRequest"`
	Sha256                   string                   `json:"sha256"`
	GUID                     string                   `json:"guid"`
	SessionID                string                   `json:"session_id"`
	VerdictRequestAttributes VerdictRequestAttributes `json:"verdict_request_attributes"`
	UseCache                 bool                     `json:"use_cache"`
	UseHashLookup            bool                     `json:"use_shed"`
}

// GetGUID returns the GUID of the verdictRequest.
func (r verdictRequest) GetGUID() string {
	return r.GUID
}

// NewVerdictRequest creates a new verdictRequest instance.
func NewVerdictRequest(sessionID string, options options.VaasOptions, sha256 string) VerdictRequest {
	return verdictRequest{
		Kind:          VerdictRequestKind,
		Sha256:        sha256,
		SessionID:     sessionID,
		GUID:          uuid.New().String(),
		UseCache:      options.UseCache,
		UseHashLookup: options.UseHashLookup,
	}
}

// NewVerdictRequestWithAttributes creates a new verdictRequest instance with attributes.
func NewVerdictRequestWithAttributes(sessionID string, options options.VaasOptions, sha256 string, attributes VerdictRequestAttributes) VerdictRequest {
	return verdictRequest{
		Kind:                     VerdictRequestKind,
		Sha256:                   sha256,
		SessionID:                sessionID,
		GUID:                     uuid.New().String(),
		UseCache:                 options.UseCache,
		UseHashLookup:            options.UseHashLookup,
		VerdictRequestAttributes: attributes,
	}
}
